### PR TITLE
Fix Detected both log4j-over-slf4j.jar AND slf4j-log4j12.jar on the class path, preempting StackOverflowError.

### LINKED ISSUE #2405: Detected both log4j-over-slf4j.jar AND slf4j-log4j12.jar on the class path, preempting StackOverflowError.
I find that bk depends on log4j-over-slf4j and slf4j-log4j12, sometimes this will cause error. 
http://www.slf4j.org/codes.html#log4jDelegationLoop

### PR DESCRIPTION
**Pull request by Technoboy-**
Fix #2405 .

### CODE REVIEW

**Comment by eolivelli** [comment]
Did it happen only on microbenchmarks ?
Interesting

**Comment by Technoboy-** [comment]
> Did it happen only on microbenchmarks ?
> Interesting

Yes.  And if we start bookie locally, it will throw the error.

**Comment by eolivelli** [comment]
I am not sure I follow, the microbenchmarks is not intended to have an impact on the bookie service.
Do you have a stacktrace to share ?

**Comment by Technoboy-** [comment]
> I am not sure I follow, the microbenchmarks is not intended to have an impact on the bookie service.
> Do you have a stacktrace to share ?

From this [wiki](http://www.slf4j.org/codes.html#log4jDelegationLoop),  it suggests not to keep the two jar together.
If we run Bookie locally, two jars are all in the classpath.

**Comment by eolivelli** [comment]
You change does not affect the list of jars on the classpath, because you modified only the 'microbenchmarks' module, that is not a source for the 'bookie' service.
In fact we have an automatic check that checks every jar in the binary tarball and detects if there is a companion LICENSE file.

If your change drops a jar from the classpath of the bookie (and so it is no more present in the .tar.gz file) I expect that check to fail because we did not update the list of LICENSE files.

Can you please verify in the two .tar.gz files created inside bookkeeper-dist/server and bookkeeper-dist/all modules that the jar was present before the patch and it is now no more  present ?

**Comment by Technoboy-** [comment]
> You change does not affect the list of jars on the classpath, because you modified only the 'microbenchmarks' module, that is not a source for the 'bookie' service.

I have removed the dependency of log4j-over-slf4j.jar in the parent pom.

File: pom.xml [hunk 1]
```diff
@@ -251,11 +251,6 @@
         <artifactId>slf4j-log4j12</artifactId>
         <version>${slf4j.version}</version>
       </dependency>
-      <dependency>
```
Comment (eolivelli) [inline pom.xml:254]: this is inside "<dependencyManagement>" section, it it only a declaration of the version, it is not using the jar
see
https://stackoverflow.com/questions/2619598/differences-between-dependencymanagement-and-dependencies-in-maven

Comment (Technoboy-) [reply pom.xml:254]: Yes, I know.  But it's really occurs in my local and do not reproduce after retry.
Well, it is also ambiguous to declare the two.

Comment (Technoboy-) [reply pom.xml:254]: Should I close this pr ?

**Comment by Technoboy-** [comment]
Should I close this pr ?

**Pull request closed without merging**